File: DTC.Api/Dtos/Tournament/TournamentDto.cs
using DTC.Api.Dtos.Group;
using DTC.Api.Dtos.Braket;
using DTC.Api.Dtos.TournamentPlayer;
using DTC.Api.Enums;

namespace DTC.Api.Dtos.Tournament
{
    public class TournamentDto
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }

        public TournamentStatus Status { get; set; }

        public DateTimeOffset StartDate { get; set; }
    }
}

File: dtc.frontend/src/dtos/Tournament/TournamentDto.ts
import z from "zod";
import { roundDtoSchema } from "../Round/RoundDto";
import { TournamentMode } from "../../enums/TournamentMode";
import { TournamentStatus } from "../../enums/TournamentStatus";

export const tournamentDtoSchema = z.object({
  id: z.number(),
  name: z.string(),
  startDate: z.iso.datetime({ offset: true }),
  description: z.string().nullable(),
  mode: z.enum(TournamentMode),
  status: z.enum(TournamentStatus),
  rounds: z.array(roundDtoSchema),
});

export type TournamentDto = z.infer<typeof tournamentDtoSchema>;
